testplan/testing/multitest/base.py

```diff
@@ -988,17 +1008,22 @@ def _run_parallel_testcases(self, testsuite, execution_groups):
                 # If any testcase errors and we are configured to stop on
                 # errors, we still wait for the rest of the current execution
                 # group to finish before stopping.
-                if (
-                    testcase_report.status == Status.ERROR
-                    and self.cfg.stop_on_error
-                ):
-                    should_stop = True
+                if testcase_report.status == Status.ERROR:
+                    if self.cfg.stop_on_error:
+                        self.logger.debug(
+                            'Stopping execution of testsuite "%s" due to error.',
+                            testsuite.name,
+                        )
+                        should_stop = True
+                elif testcase_report.status == Status.FAILED:
```

Comment:
Also I think the expected behavior is if we are executing testcases in parallel, and when error/failure occurs, testcases that have not started shall cancel.